app/Http/Controllers/ProjectController.php

```diff
@@ -7,156 +7,195 @@
 use App\Project;
 use App\Member;
 
+
 class ProjectController extends Controller
 {
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index(Request $request)
-    {
-        if ($request->get('pid')) {
-            $projects = Project::find(16);
-            $projects->members->pluck('id')->toArray();
-        }
-        $projects = Project::all();
-        $messages = [];
-        $action = $request->get('ACTION');
-        switch($action) {
-            case 1: {
-                $messages[] = 'Create project successfully !';
-                break;
+        /**
+         * Display a listing of the resource.
+         *
+         * @return \Illuminate\Http\Response
+         */
+        public function index(Request $request)
+        {
+            $members = [];
+            if ($request->get('pid')) {
+                $projects = Project::find($request->get('pid'));
+                $members = $projects->members->pluck('id')->toArray();
             }
-            case 2: {
-                $messages[] = 'Update project successfully !';
-                break;
+            $projects = Project::all();
+            $messages = [];
+            $action = $request->get('ACTION');
+            switch($action) {
+                case 1: {
+                    $messages[] = 'Create project successfully !';
+                    break;
+                }
+                case 2: {
+                    $messages[] = 'Update project successfully !';
+                    break;
+                }
+                default: {
+                    break;
+                }
             }
-            default: {
-                break;
+            $result = [
+            'projects' => $projects,
+            'messages' => $messages,
+            'members' => $members
+            ];
+            return response()->json($result);
+        }
+
+        /**
+         * Store a newly created resource in storage.
+         *
+         * @param  \Illuminate\Http\Request  $request
+         * @return \Illuminate\Http\Response
+         */
+        public function store(Request $request)
+        {
+            $request->validate(self::RULES);
+            $project = new Project([
+                'name' => $request->get('name'),
+                'information' => $request->get('information'),
+                'deadline' => $request->get('deadline'),
+                'type' => $request->get('type'),
+                'status' => $request->get('status')
+                ]);
+            $save = $project->save();
+            if ($save) {
```

Comment:
This is good. Can you do the same for the other controllers and controller functions?

Reply:
I have do it